pandas/core/indexes/multi.py

```diff
@@ -563,6 +563,20 @@ def from_tuples(
             arrs = zip(*tuples)
             arrays = cast(List[Sequence[Hashable]], arrs)
 
+        if all(isinstance(e, tuple) for e in tuples):
+            if not all(tuples):
```

Comment:
Doesn't the above condition satisfy this one?

Reply:
No. `tuples` can be not list of tuples. For example, based on a test case,  `tuples` can be 
```
tuples = [array([0, 1], dtype=int64), array([0, 0], dtype=int64), array([1, 0], dtype=int64)]
sortorder = None, names = ['2', 'a']
```
from `pandas/tests/strings/test_api.py` , `test_api_per_method` function